hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/statemachine/background/BlockDeletingService.java

```diff
@@ -432,6 +430,8 @@ public ContainerBackgroundTaskResult deleteViaSchema2(
               meta.getStore().getBlockDataTable().deleteWithBatch(batch, bID);
             }
           }
+          totalBlocks =
+              deleteTransactions(delBlocks, handler, blockDataTable, container);
           meta.getStore().getBatchHandler().commitBatchOperation(batch);
```

Comment:
thanks @cchenax for this work.
but I think `meta.getStore().getBatchHandler().commitBatchOperation(batch);` is which the metadata is definitely be deleted, thus this should be executed before `deleteTransactions(delBlocks, handler, blockDataTable, container);` , so that we can make sure metadata is deleted before deleting the corresponding blocks.